source/Higgs/UntruncatedAtMostTwo.lagda

```diff
@@ -0,0 +1,244 @@
+J. A. Carr 8 July 2025.
+
+The untruncated form of the at-most-two elements is equivalent
+to the statement that Aut Ω has exactly 1 or 2 elements
+(and hence every element is either identity or negation)
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import MLTT.Spartan
+open import MLTT.Plus-Properties
+open import UF.DiscreteAndSeparated
+open import TypeTopology.SigmaDiscreteAndTotallySeparated
+open import UF.Equiv hiding (_≅_)
+open import UF.FunExt
+open import UF.Logic
+open import UF.PropTrunc
+open import UF.Subsingletons
+open import UF.SubtypeClassifier hiding (Ω)
+open import Fin.Type
+open import Fin.Topology
+open import Fin.Pigeonhole
+
+module Higgs.UntruncatedAtMostTwo
+        {𝓤 : Universe}
+        (fe : Fun-Ext)
+        (pe : propext 𝓤)
+        (pt : propositional-truncations-exist)
+       where
+
+open import Higgs.Rigidity fe pe
+open import Higgs.InvolutionTheorem fe pe
+open import Higgs.AutomorphismsOfOmega fe pe
+open import Higgs.AutomorphismsOfOmegaWEM fe pe pt
+
+open Conjunction
+open Implication fe
+open Universal fe
+
+\end{code}
+
+Assuming function extensionality, having an untruncated pigeonhole principle
+reflects discreteness to the codomain.
+
+The core method is to note that the pidgeonhole function must respect equality
+of functions, so we produce a pair of functions, where equality of their results
+holds if and only if the two elements of the codomain are equal.
+
+\begin{code}
+
+constantly : {𝓤 𝓥 : Universe}
+           → {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+           → Y → X → Y
+constantly const-y _ = const-y
+
+almost-constantly : {𝓤 𝓥 : Universe}
+                  → {X : 𝓤 ̇ } {Y : 𝓥 ̇ }
+                  → is-discrete X
+                  → X → Y → Y → X → Y
+almost-constantly X-discrete pt-x pt-y const-y x' =
```

Comment:
Please call these `x` and `y` and `y'` rather than what you have. The current names are confusing.

Reply:
I think I originally had x x' y y', but no matter which way these are oriented I find it confusing and errorprone. Changed nonetheless